WordPress/Classes/ViewRelated/Post/FakePreviewBuilder.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+
+class FakePreviewBuilder: NSObject {
+    let title: String?
+    let content: String?
+    let tags: [String]
+    let categories: [String]
+    let message: String?
+
+    init(title: String?, content: String?, tags: [String], categories: [String], message: String?) {
+        self.title = title
+        self.content = content
+        self.tags = tags
+        self.categories = categories
+        self.message = message
+        super.init()
+    }
+
+    func build() -> String {
+        let template = loadTemplate()
+        let messageParagraph = message.map({ "<p>\($0)</p>" }) ?? ""
+        return template
+            .replacingOccurrences(of: "!$title$!", with: previewTitle)
+            .replacingOccurrences(of: "!$text$!", with: previewContent)
+            .replacingOccurrences(of: "!$mt_keywords$!", with: previewTags)
+            .replacingOccurrences(of: "!$categories$!", with: previewCategories)
+            .replacingOccurrences(of: "<div class=\"page\">", with: "<div class=\"page\">\(messageParagraph)")
+    }
+
+    private func loadTemplate() -> String {
+        guard let path = Bundle.main.path(forResource: "defaultPostTemplate", ofType: "html"),
+            let template = try? String(contentsOfFile: path, encoding: .utf8) else {
+                assertionFailure("Unable to load preview template")
+                return ""
+        }
+        return template
+    }
+}
+
+// MARK: - Formatting Fields
+
+private extension FakePreviewBuilder {
+    var previewTitle: String {
+        return title?.nonEmptyString() ?? NSLocalizedString("(no title", comment: "")
```

Comment:
Maybe we could add some comment text to benefit our polyglots?  Here and elsewhere.